chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/surveillance/AddDocumentActionPermissions.java

```diff
@@ -20,11 +36,35 @@ public boolean hasAccess(Object obj) {
         } else if (getResourcePermissions().isUserRoleAdmin() || getResourcePermissions().isUserRoleOnc()) {
             return true;
         } else if (getResourcePermissions().isUserRoleAcbAdmin()) {
-            Long acb = (Long) obj;
-            return isAcbValidForCurrentUser(acb);
+            Long nonconformityId = (Long) obj;
+            try {
+                SurveillanceEntity surv = survDao.getSurveillanceByNonconformityId(nonconformityId);
+                return isAcbValidForCurrentUser(surv.getCertifiedProduct().getCertificationBodyId())
+                        && !isNonconformityForRemovedCriteria(surv, nonconformityId);
+            } catch (EntityRetrievalException ex) {
+                return false;
+            }
         } else {
             return false;
         }
     }
 
+    private boolean isNonconformityForRemovedCriteria(SurveillanceEntity surv, Long nonconformityId) {
+        if (!ff4j.check(FeatureList.EFFECTIVE_RULE_DATE_PLUS_ONE_WEEK)) {
+            return false;
```

Comment:
I know the ticket is flagged with this flag, but should security care? If there were a removed criteria today we'd still want to restrict ACBs from adding/removing documents related to that criteria, right?